### DISCUSSION TITLE
Multi-instance learning heat map attention maps and self-attention matrices

### DISCUSSION THREAD

**Discussion by yarkinslav** [Project-MONAI/tutorials#1512] [Q&A] [answered]
Hi, thanks for great tutorial and discussion guides!
1) I am trying to get intuition of how transformer is able to learn dependencies across instances of bag, and wondering if it possible to extract self-attention matrices, product of Q and K from Transfomer Encoder Layer?
2) Is it possible to visualize the heat maps of prediction (weighted by attention layer), e.g. some analogy of grad cam in CNNs, but in output of transformer encoder?

Great thanks in advance!

**Comment by wyli** [comment]
Hi, the 2nd point might be related to https://github.com/Project-MONAI/MONAI/discussions/5152

**Reply by yarkinslav** [reply, marked as answer]
Thanks for reply an the link to previous discussion, however, it’s not clear how attention weight could be achieved by just setting no_head=false, since it performs the transformer encoding and set to false as default. More likely, we can output the weights by printing a, on the attention step:
https://github.com/Project-MONAI/MONAI/blob/ca90628ffbadf37031b06d2506c7938c0a29f25b/monai/networks/nets/milmodel.py#L222
